crates/cast/bin/cmd/storage.rs

```diff
@@ -202,23 +203,70 @@ async fn fetch_storage_slots(
     address: NameOrAddress,
     layout: &StorageLayout,
 ) -> Result<Vec<B256>> {
-    // TODO: Batch request
-    let futures: Vec<_> = layout
-        .storage
-        .iter()
-        .map(|slot| {
-            let slot = B256::from(U256::from_str(&slot.slot)?);
-            Ok(provider.get_storage_at(address.clone(), slot.to_ethers(), None))
-        })
-        .collect::<Result<_>>()?;
-
-    join_all(futures).await.into_iter().map(|r| Ok(r?.to_alloy())).collect()
+    let mut futures = Vec::new();
+
+    for storage_entry in &layout.storage {
+        let slot_number = U256::from_str_radix(&storage_entry.slot, 10)?.to_be_bytes();
+        let future =
+            provider.get_storage_at(address.clone(), B256::from(slot_number).to_ethers(), None);
+        futures.push(future);
+    }
+
+    let results = join_all(futures).await;
```

Comment:
if we don't need to change anything here, then let's use the previous code

Reply:
> if we don't need to change anything here, then let's use the previous code

hey, sorry for the late, just got free time, we can't use first code since we can't collect data from a vector of future into Result<Vec<>,>, since Result is not async, let me know if i need to change somethihng ?